mwdb/model/file.py

```diff
@@ -74,17 +76,35 @@ def get_or_create(cls, file, parent=None, metakeys=None, share_with=None):
 
         if is_new:
             file.stream.seek(0, os.SEEK_SET)
-            file.save(file_obj.get_path())
+            print(app_config.mwdb.storage_provider)
+            if app_config.mwdb.storage_provider == StorageProviderType.BLOB:
+                print(file_obj.get_path())
+                s3 = boto3.resource('s3',
+                    endpoint_url=app_config.mwdb.blob_storage_endpoint,
+                    aws_access_key_id=app_config.mwdb.blob_storage_access_key,
+                    aws_secret_access_key=app_config.mwdb.blob_storage_secret_key,
+                    config=BotoConfig(signature_version='s3v4'),
+                    region_name=app_config.mwdb.blob_storage_region_name)
+                s3.Bucket(app_config.mwdb.blob_storage_bucket_name).upload_fileobj(file, file_obj.get_path())
+            else:
+                file.save(file_obj.get_path())
 
         return file_obj, is_new
 
     def get_path(self):
-        upload_root = app_config.mwdb.uploads_folder
+        # Some S3 "compatible" implementations don't suppose an initial slash
+        # so if you want to upload into the root you need a blank path
+        upload_path = app_config.mwdb.uploads_folder or ""
```

Comment:
I think the comment is out of date, my bad - basically if you have the upload_path set to / then it'll blow up because even though you're technically wanting to upload into the root of the bucket (if it's set to /) you get an error due to MinIO not supporting an initial slash in the filename (idk why, seems dumb but oh well). I noticed that if I set the uploads_folder to nothing in the config that it would be None, which would blow up when we use `os.path.join(upload_path, sample_sha256)`. 

> None or "" => "" ok, None is changed to ""
✔️ - None is not supported
> "" or "" => "" ok, "" is unchanged
✔️ - "" is assumed to be root
> "/" or "" => "/" slash stays a slash?
✔️ - This probably works on S3, but on MinIO it will blow up - but the exception is pretty obvious
> "uploads/123" or "" => "uploads/123" ok, dirnames without a leading slash work
✔️ - I think os.path.join should handle this ok
> "/root" or "" => "/root" leading slash is kept?
✔️ - I think a leading slash is fine as long as you're referencing a "directory"

I'll fix the comment when I'm at my work machine but do you know if it's indended that "" becomes None in the typed-config thing?

Reply:
thanks for your clarification!